src/lib/ensureSSL.ts

```diff
@@ -0,0 +1,17 @@
+/*
+* Makes sure that any http requests get redirected to https
+*/
+
+import { parse } from "url"
+const { APP_URL } = process.env
+
+export const ensureSSL = (req, res, next) => {
+  const protocol = req.protocol
+  const isSecure = req.secure || protocol === "https:"
+
+  if (!isSecure && parse(APP_URL as string).protocol === "https:") {
+    res.redirect(301, APP_URL + req.url)
```

Comment:
Would it make sense to construct the new URL by replacing the protocol of the current URL with `https:` instead of looking to a .env variable (which we could potentially forget to set)?

Reply:
I think `APP_URL` is something we should always be setting. But the reason I went with this approach was because generally we don't require https for local / dev environments (we also don't have it set up - we get the error `This site can’t provide a secure connection` when trying to navigate to https://localhost:4000/playground). 

This allows us to connect via https in staging and production environments but not require it locally.

Reply:
Makes sense - thank you!